### PR TITLE
`Iris`: Allow team repository access for Iris

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Iris should be able to help on all programming exericses; however, it currently can't see the code of submissions in team exercises.

### Description

Added compatibility with team exercises by checking wether the exercise is a team exercise, then altering the retrieval logic accordingly.

### Steps for Testing

Prerequisites:
- 1 Team Programming Exercise
- 2 Students in a team within that exercise

1. Submit something to the programming exercise
2. Talk to your submitted code with Iris using both students accounts
3. Verify that the student who did not submit can also talk to Iris about the code submitted by student 1

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review

- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance even for very large courses with more than 2000 students.
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance even for very large courses with more than 2000 students.
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2
#### Performance Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| IrisExerciseChatSessionService.java | 93% | ✅                           |
| ProgrammingExerciseStudentParticipationRepository.java | 80% | ✅              |

## Summary by CodeRabbit

- **New Features**
	- Enhanced retrieval of student participations for team exercises.
	- Added a method to fetch participations based on team and exercise ID.
	- New functionality to mock HTTP requests for programming exercise chat responses.
	- Introduced a method to get the first team exercise of a specified type.

- **Bug Fixes**
	- Improved error handling and structure in integration tests for solo and team exercises.

- **Tests**
	- Refactored integration tests to differentiate between solo and team exercises, enhancing clarity and organization.

### CODE REVIEW

File: src/test/java/de/tum/cit/aet/artemis/core/connector/IrisRequestMockProvider.java [hunk 1]
```diff
@@ -100,6 +100,38 @@
...
+                }
+
+                if (!jsonNode.get("submission").has("id")) {
+                    throw new AssertionError("Field 'id' is missing in 'submission'");
+                }
+
+                long id = jsonNode.get("submission").get("id").asLong();
+                if (id != submissionId) {
+                    throw new AssertionError("Field 'id' in 'submission' is not equal to " + submissionId);
+                }
```
[the remark this answers is not in the document]
Comment (bassner) [reply src/test/java/de/tum/cit/aet/artemis/core/connector/IrisRequestMockProvider.java]: Done in [79c78e4](https://github.com/ls1intum/Artemis/pull/9975/commits/79c78e4d0380810a36f4209ccd2f20c23211217e)

**Review by HawKhiem — CHANGES_REQUESTED** [review]
I tested on TS3. I created this exercise:

https://artemis-test3.artemis.cit.tum.de/courses/29/exercises/825

I created a team of Test User 1 and Test User 2.
I used Test User 1 to submit the code and talking to Iris works fine for Test User 1. 
However when I switch to Test User 2 and tried to talk to Iris, it said that Iris is disabled for this exercise
Screenshot 2024-12-09 224556
Screenshot 2024-12-09 224727
Screenshot 2024-12-09 224743

**Comment by bassner** [comment]
I can not reproduce the issue above for test user 2.
CleanShot 2024-12-10 at 13 33 06@2x

**Review by isabellagessl — APPROVED** [review]
tested in test server 3

**Review by kaancayli — APPROVED** [review]
Tested on TS3 works as expected. Code LGTM

**Pull request merged**